tests/unit-test/test_scylladb.c

```diff
@@ -28,6 +30,105 @@ static struct identity_s {
+
+  hash8019_queue_free(&res);
+  printf("test_db_get_trytes done\n");
+}
+
+void test_db_get_approvees(db_client_service_t* service) {
+  hash243_queue_t res = NULL;
+  db_get_approvee(service, &res, iota_tx1.attachment.trunk);
+  db_get_approvee(service, &res, iota_tx1.attachment.branch);
+  db_get_approvee(service, &res, iota_tx2.attachment.trunk);
+  db_get_approvee(service, &res, iota_tx2.attachment.branch);
+
+  hash243_queue_t itr = res;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx1.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+  itr = itr->next;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx1.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+  itr = itr->next;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx2.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+  itr = itr->next;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx2.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+
+  hash243_queue_free(&res);
+  printf("test_db_get_approvee done\n");
+}
+
+void test_db_get_bundle(db_client_service_t* service) {
+  hash243_queue_t res = NULL;
+
+  db_get_transactions_by_bundle(service, &res, iota_tx1.essence.bundle);
+  db_get_transactions_by_bundle(service, &res, iota_tx2.essence.bundle);
+  hash243_queue_t itr = res;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx1.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+  itr = itr->next;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx2.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+
+  hash243_queue_free(&res);
+  printf("test_db_get_bundle done\n");
+}
+
+void test_db_get_address(db_client_service_t* service) {
+  hash243_queue_t res = NULL;
+
+  db_get_transactions_by_address(service, &res, iota_tx1.essence.address);
+  db_get_transactions_by_address(service, &res, iota_tx2.essence.address);
+  hash243_queue_t itr = res;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx1.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+  itr = itr->next;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx2.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+
+  hash243_queue_free(&res);
+  printf("test_db_get_address done\n");
+}
+
+void test_db_get_tag(db_client_service_t* service) {
+  hash243_queue_t res = NULL;
+
+  db_get_transactions_by_tag(service, &res, iota_tx1.essence.obsolete_tag);
+  db_get_transactions_by_tag(service, &res, iota_tx2.essence.obsolete_tag);
+  hash243_queue_t itr = res;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx1.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+  itr = itr->next;
+  TEST_ASSERT_EQUAL_MEMORY(iota_tx2.consensus.hash, itr->hash, sizeof(flex_trit_t) * NUM_FLEX_TRITS_HASH);
+
+  hash243_queue_free(&res);
+  printf("test_db_get_tag done\n");
+}
+
+void test_permanode(void) {
+  db_client_service_t service;
+  service.host = strdup(host);
+  TEST_ASSERT_EQUAL_INT(db_client_service_init(&service, DB_USAGE_NULL), SC_OK);
```

Comment:
Do you mind to exchange the position of these two arguments?
For unity test framework, the first argument is `expected` and the second one is `actual`

Reply:
OK!